### PR TITLE
implementation (Languages): #28476 Support the different Menu actions in the table. 

### LINKED ISSUE #28476: Support the different Menu actions in the table.
### Parent Issue

#28474

### User Story

As a user, I require the ability to view supported workflows for each language listed in the table. This enhancement will provide valuable insights into the available workflows associated with each language

### Acceptance Criteria

**Edit will be part of other taks**

- [ ] The listing table should include a column (Menu) displaying the supported workflows for each language 
image

### Proposed Objective

User Experience

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* add the menu options to the locale table and the confirmation messages. 
* Push Publish 
* Set As Default 
* Delete 

### Screenshots

https://github.com/dotCMS/core/assets/31667212/646d62f0-b6d3-4f8d-b9e4-173e955160b8

### CODE REVIEW

File: core-web/libs/data-access/src/lib/dot-languages/dot-languages.service.spec.ts [hunk 1]
```diff
@@ -10,18 +15,65 @@ describe('DotLanguagesService', () => {
...
+    it('should make a language the default language', () => {
+        const id = 1;
+        spectator.service.makeDefault(id).subscribe();
+        spectator.expectOne(`${LANGUAGE_API_URL}/${id}/_makedefault`, HttpMethod.PUT);
+    });
+
+    // it('should get languages and countries in ISO format', () => {
+    //     spectator.service.getISO().subscribe();
+    //     spectator.expectOne(`${LANGUAGE_API_URL}/iso`, HttpMethod.GET);
+    // });
```
Comment (zJaaal) [inline core-web/libs/data-access/src/lib/dot-languages/dot-languages.service.spec.ts:78]: should we remove this?

File: core-web/libs/dotcms-models/src/lib/dot-language.model.ts [hunk 2]
```diff
@@ -7,4 +7,10 @@ export interface DotLanguage {
     defaultLanguage?: boolean;
     translated?: boolean;
     isoCode?: string;
+    variables?: { count: number; total: number };
+}
+
+export interface DotLanguagesISO {
+    countries: { code: string; name: string }[];
+    languages: { code: string; name: string }[];
```
Comment (zJaaal) [inline core-web/libs/dotcms-models/src/lib/dot-language.model.ts:15]: can we make `{ code: string; name: string }` an interface?

File: core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.spec.ts [hunk 3]
```diff
@@ -45,14 +88,41 @@ describe('DotLocalesListStore', () => {
...
+        expect(messageService.add).toHaveBeenCalled();
+    });
+
+    it('should set the local actions correctly', (done) => {
+        spectator.service.setLocales([...mockLocales]);
 
         spectator.service.vm$.subscribe((viewModel) => {
-            expect(viewModel.locales.length).toBe(0);
+            const defaultLocale = viewModel.locales[0];
+            const notDefaultLocale = viewModel.locales[1];
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.spec.ts:116]: check the lengths too so if we add more actions this test will fail

File: core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts [hunk 4]
```diff
@@ -16,64 +39,232 @@ export interface DotLocaleRow {
...
 }
 
-export interface DotLocaleListViewModel {
+export interface DotLocaleListViewModel extends DotLanguagesISO {
     locales: DotLocaleRow[];
+    initialLocales: DotLanguage[];
+}
+
+export interface DotLocaleListResolverData {
+    localaes: DotLanguage[];
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts:56]: you have a typo here and this interface looks like an extension of an interface you declared in another file

File: core-web/libs/portlets/dot-locales/portlet/src/lib/share/ui/DotLocaleConfirmationDialog/DotLocaleConfirmationDialog.component.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+export interface DotLocaleConfirmationDialogData {
+    acceptLabel: string;
+    icon: string;
+    ISOCode: string;
+    locale: DotLanguage;
+    message: string;
+}
+
+@Component({
+    selector: 'dot-dot-locale-confirmation-dialog',
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-locales/portlet/src/lib/share/ui/DotLocaleConfirmationDialog/DotLocaleConfirmationDialog.component.ts:21]: typo here `dot-dot`

**Review by zJaaal — DISMISSED** [review]

**Review by rjvelazco — CHANGES_REQUESTED** [review]

File: core-web/libs/portlets/dot-locales/portlet/src/lib/share/ui/DotLocaleConfirmationDialog/DotLocaleConfirmationDialog.component.html
Comment (fmontes) [inline core-web/libs/portlets/dot-locales/portlet/src/lib/share/ui/DotLocaleConfirmationDialog/DotLocaleConfirmationDialog.component.html:1]: Why this is not a `p-dialog`?

File: core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts [hunk 6]
```diff
@@ -16,64 +39,232 @@ export interface DotLocaleRow {
     actions: DotActionMenuItem[];
 }
 
-export interface DotLocalesListState {
+export interface DotLocalesListState extends DotLanguagesISO {
     status: ComponentStatus;
     locales: DotLocaleRow[];
+    initialLocales: DotLanguage[];
```
Comment (fmontes) [inline core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts:45]: What is the diff between `locales` and `initialLocales` and whay are diff types?

File: core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts [hunk 7]
```diff
@@ -16,64 +39,232 @@ export interface DotLocaleRow {
...
+    ): DotLocaleRow[] {
+        return locales.map((locale) => ({
+            id: locale.id,
+            locale: `${locale.language} (${locale.isoCode})`,
+            language: `${locale.language} - ${locale.languageCode}`,
+            country: `${locale.country} - ${locale.countryCode}`,
             variables: 'TBD',
-            defaultLanguage: language.defaultLanguage,
+            defaultLanguage: locale.defaultLanguage,
             actions: [
```
Comment (fmontes) [inline core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts:177]: This seems like a lot of state here... should we this be his own state and component? We had this problem in Pages, try to add all the actions of the menus in the same store as the page and now is a big spagetti code.

**Review by fmontes — DISMISSED** [review]

File: core-web/libs/data-access/src/lib/dot-languages/dot-languages.service.ts [hunk 8]
```diff
@@ -40,4 +44,95 @@ export class DotLanguagesService {
...
+     *
+     * @param {{
+     *          language: string;
+     *         languageCode: string;
+     *         country?: string;
+     *         countryCode?: string;
+     *     }} language - The language to be added.
+     * @return {Observable<DotLanguage>} An observable of the language added.
+     */
+    add(language: {
```
Comment (oidacra) [inline core-web/libs/data-access/src/lib/dot-languages/dot-languages.service.ts:59]: Convert this to a type

File: core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts [hunk 9]
```diff
@@ -16,64 +39,232 @@ export interface DotLocaleRow {
...
+                    detail: this.dotMessageService.get('locale.notification.success.message')
+                });
+            }),
+            catchError((error: HttpErrorResponse) => this.dotHttpErrorManagerService.handle(error))
+        );
+    });
+
+    readonly deleteLocale = this.effect<number>((languageId$) =>
+        languageId$.pipe(
+            switchMap((languageId) => this.languageService.delete(languageId)),
```
Comment (oidacra) [inline core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts:132]: What happend if the network is really slow? in the UI, do you have any `Loading` status?

File: core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.spec.ts (hunk 3 above)
Comment (hmoreras) [reply core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.spec.ts:116]: good catch. ! done

File: core-web/libs/data-access/src/lib/dot-languages/dot-languages.service.ts [hunk 10]
```diff
@@ -40,4 +44,95 @@ export class DotLanguagesService {
...
+        language: string;
+        languageCode: string;
+        country?: string;
+        countryCode?: string;
+    }): Observable<DotLanguage> {
```
[the remark this answers is not in the document]
Comment (hmoreras) [reply core-web/libs/data-access/src/lib/dot-languages/dot-languages.service.ts:64]: not really... the DotLanguage is the one, but these are the onlyones needed for creation.

File: core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts (hunk 4 above)
Comment (hmoreras) [reply core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts:56]: removed

File: core-web/libs/data-access/src/lib/dot-languages/dot-languages.service.ts (hunk 8 above)
Comment (hmoreras) [reply core-web/libs/data-access/src/lib/dot-languages/dot-languages.service.ts:59]: done

File: core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts (hunk 7 above)
Comment (hmoreras) [reply core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts:177]: I don't think in this case; languages only have a max of 4 actions.

File: core-web/libs/dotcms-models/src/lib/dot-language.model.ts (hunk 2 above)
Comment (hmoreras) [reply core-web/libs/dotcms-models/src/lib/dot-language.model.ts:15]: done

File: core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts [hunk 11]
```diff
@@ -16,64 +39,232 @@ export interface DotLocaleRow {
...
-    readonly setLocales = this.updater((state: DotLocalesListState, languages: DotLanguage[]) => ({
+
+    readonly setResolvedData = this.updater(
+        (
+            state: DotLocalesListState,
+            routeResolved: {
+                data: DotLocalesListResolverData;
+                isEnterprise: boolean;
+                pushPublishEnvironments: DotEnvironment[];
+            }
```
[the remark this answers is not in the document]
Comment (hmoreras) [reply core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts:81]: implementation changed.

File: core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts (hunk 6 above)
Comment (hmoreras) [reply core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts:45]: good calll, actually wont need the `initialLocales` ( raw data from endpoint ), removed. `locales` is the data ready to be presented in the UI.

File: core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts (hunk 9 above)
Comment (hmoreras) [reply core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts:132]: Will work loading UI in other PR, but add the  state changed in the store

File: core-web/libs/portlets/dot-locales/portlet/src/lib/share/ui/DotLocaleConfirmationDialog/DotLocaleConfirmationDialog.component.html
Comment (hmoreras) [reply core-web/libs/portlets/dot-locales/portlet/src/lib/share/ui/DotLocaleConfirmationDialog/DotLocaleConfirmationDialog.component.html:1]: is called by the prime ng Dynamic Dialog.

File: core-web/libs/portlets/dot-locales/portlet/src/lib/share/ui/DotLocaleConfirmationDialog/DotLocaleConfirmationDialog.component.ts [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    templateUrl: './DotLocaleConfirmationDialog.component.html',
+    styleUrl: './DotLocaleConfirmationDialog.component.scss',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DotLocaleConfirmationDialogComponent {
+    readonly config: DynamicDialogConfig<DotLocaleConfirmationDialogData> =
+        inject(DynamicDialogConfig);
+    ref = inject(DynamicDialogRef);
+
+    data: DotLocaleConfirmationDialogData = this.config.data || {
```
Comment (rjvelazco) [inline core-web/libs/portlets/dot-locales/portlet/src/lib/share/ui/DotLocaleConfirmationDialog/DotLocaleConfirmationDialog.component.ts:33]: Should this be in `ngOnInit`?

**Review by rjvelazco — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

File: core-web/libs/portlets/dot-locales/portlet/src/lib/share/ui/DotLocaleConfirmationDialog/DotLocaleConfirmationDialog.component.ts (hunk 12 above)
Comment (hmoreras) [reply core-web/libs/portlets/dot-locales/portlet/src/lib/share/ui/DotLocaleConfirmationDialog/DotLocaleConfirmationDialog.component.ts:33]: config is defined with the `inyect` so looks like not.

**Review by zJaaal — APPROVED** [review]

**Pull request merged**